src/Control/Carrier/Parse/Simple.hs

```diff
@@ -23,23 +23,21 @@ import           Parsing.Parser
 import           Parsing.TreeSitter
 
 runParse :: Duration -> ParseC m a -> m a
-runParse timeout = runReader timeout . runParseC
+runParse timeout (ParseC m) = runReader timeout m
 
-newtype ParseC m a = ParseC { runParseC :: ReaderC Duration m a }
+newtype ParseC m a = ParseC (ReaderC Duration m a)
```

Comment:
We no longer define or export record selectors for carriers unless the carrier amounts to `newtype ThingC m a = ThingC { runThing :: m a }`, or are otherwise basically forced to. This reduces the surface area of the library a bit while also making it clearer how you are expected to use these things.